recipes/cuda-nvcc/build.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# Install to conda style directories
+[[ -d lib64 ]] && mv lib64 lib
+
+[[ ${target_platform} == "linux-64" ]] && targetsDir="targets/x86_64-linux"
+[[ ${target_platform} == "linux-ppc64le" ]] && targetsDir="targets/ppc64le-linux"
+[[ ${target_platform} == "linux-aarch64" ]] && targetsDir="targets/sbsa-linux"
+
+for i in `ls`; do
+    [[ $i == "build_env_setup.sh" ]] && continue
+    [[ $i == "conda_build.sh" ]] && continue
+    [[ $i == "metadata_conda_debug.yaml" ]] && continue
+    if [[ $i == "lib" ]] || [[ $i == "include" ]]; then
+        # Headers and libraries are installed to targetsDir
+        mkdir -p ${PREFIX}/${targetsDir}
+        mkdir -p ${PREFIX}/$i
+        cp -rv $i ${PREFIX}/${targetsDir}
+        if [[ $i == "lib" ]]; then
+            for j in "$i"/*.a*; do
+                # Shared and static libraries are symlinked in $PREFIX/lib
+                ln -sv ../${targetsDir}/$j ${PREFIX}/$j
+            done
+        fi
+    else
+        cp -r $i ${PREFIX}
```

Comment:
Some questions:
- I assume that we're intentionally not including the contents of `bin` into the `targetsDir`?
- We probably want to put the `nvvm` contents into the `targetsDir` right?
- `cicc` is inside `nvvm/bin`, but `nvcc` needs to be able to find it. Does that need special handling to be placed into `${PREFIX}/bin`?

Reply:
The current layout is fine. The `bin` and `nvvm` directories will be copied into `${PREFIX}/bin`/ and `${PREFIX}/nvvm`. Since the relative layot has stayed consistent, `nvcc` doesn't need to be modified to know where the `nvvm` directory is

Reply:
Are we OK writing to `${PREFIX}/nvvm`? My understanding was that we were avoiding writing to such top-level directories as much as possible to avoid clobbering, but perhaps we're OK treating `nvvm` as CUDA-specific in the same way that we are treating `targets` as CUDA-specific.

Do we still want to include some of these contents in `targets` or only put them in the top-level `${PREFIX}/*` directories?

Reply:
We need `nvvm` in `$PREFIX` due to the relative pathing Robert mentioned. `nvcc` expects `bin` `nvvm` and `targets` to all be at the same level. 

Clobbering shouldn't be a concern here as it was with things like `src/example.c`

Reply:
We are already patching `nvcc.profile`, so it's not a big deal to patch a couple more lines.
If we are going with filesystem hierarchy, they should not be in `$PREFIX` and rather in somewhere like `$PREFIX/lib/nvcc/nvvm` and `$PREFIX/lib/nvcc/targets` just like `gcc` does with `$PREFIX/lib/gcc`

Thoughts @mbargull, @conda-forge/core?

Reply:
`nvcc.profile` is not patched on Linux. It is only patched on Windows to address shipping different CCCL versions as requested in PR ( https://github.com/conda-forge/staged-recipes/pull/21953 ). Preferably it would not be patched at all (even on Windows). Open to a better solution that doesn't require the patch on Windows (if we have ideas). Though this is probably best taken up in the feedstock since we are blocked on Windows builds atm ( https://github.com/conda-forge/staged-recipes/pull/21723#discussion_r1160853681 ).

It's worth noting that there are already cases where we install non-traditional things in `$PREFIX`. For example [`go` adds `$PREFIX/go`]( https://github.com/conda-forge/go-feedstock/blob/865b217fa6b54e87b050e0056d61ad2faf669119/recipe/build-base.sh#L69 ). With `go`, trying to move away from that model breaks the compiler in weird ways.

Typically all CUDA components are placed in the same relative paths where they are then QA’d and deployed. The proposal here suggests dividing them up between different paths that will deviate from the process. This risks introducing new issues unique to this non-standard deployment model and makes it harder to support.

Finally would add Numba is [dependent on this location]( https://github.com/numba/numba/blob/be990a818c377408a1f41c37c7720bc2ba0fb89a/numba/cuda/cuda_paths.py#L144 ). So changing it here means Numba (and maybe other libraries) will need updates.